### PR TITLE
feat: create gmail draft message complete

### PR DESCRIPTION
**Pull request by HamoonZamiri**

### CODE REVIEW

File: langchain/src/tools/gmail/create_draft.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+  static args = [
+    { name: "message", description: "The message to include in the draft" },
+    { name: "to", description: "The list of recipients" },
+    { name: "subject", description: "The subject of the message" },
+    { name: "cc", description: "The list of CC recipients", isList: true },
+    { name: "bcc", description: "The list of BCC recipients", isList: true },
+  ];
+
+  protected prepareDraftMessage(message: string,
```
Comment (uthmanmoh) [inline langchain/src/tools/gmail/create_draft.ts]: why is this protected? did you mean private?

File: langchain/src/tools/gmail/create_draft.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            "",
+            message,
+          ].join("\n");
+
+          draftMessage.message.raw = encodeToBase64(email);
+
+          return draftMessage;
+    }
+
+    async _call(args: CreateDraftSchema) {
```
Comment (uthmanmoh) [inline langchain/src/tools/gmail/create_draft.ts]: doesnt this args param conflict with the static args field? is the static thing supposed to be a fallback if theres nothing passed in here or what? maybe they should have different names

File: langchain/src/tools/gmail/create_draft.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        const auth = await this.getAuth();
+
+        const gmail = google.gmail({ version: "v1", auth });
```
Comment (uthmanmoh) [inline langchain/src/tools/gmail/create_draft.ts]: maybe for later we should move the `gmail` object into the parent class if all children are gonna be redefining this with the exact same line

File: langchain/src/tools/gmail/create_draft.ts [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+export interface CreateDraftSchema {
+  message: string;
+  to: string[];
+  subject: string;
+  cc?: string[];
+  bcc?: string[];
+}
+
+export class GmailCreateDraft
+  extends GmailBaseTool
```
Comment (uthmanmoh) [inline langchain/src/tools/gmail/create_draft.ts]: whys this on a new line? did the linter do that? the line doesnt look that long to me

**Review by uthmanmoh — CHANGES_REQUESTED** [review]

File: langchain/src/tools/gmail/create_draft.ts (hunk 2 above)
Comment (HamoonZamiri) [reply langchain/src/tools/gmail/create_draft.ts]: The python to JS translator turned 
image
 into the static args array, not sure why this is useful in either repository. Should we remove it?

File: langchain/src/tools/gmail/create_draft.ts (hunk 1 above)
Comment (HamoonZamiri) [reply langchain/src/tools/gmail/create_draft.ts]: I agree, makes more sense to be private.

File: langchain/src/tools/gmail/create_draft.ts (hunk 2 above)
Comment (uthmanmoh) [reply langchain/src/tools/gmail/create_draft.ts]: Doesn't seem like we're using the static args anyways right? if so then yea lets remove it

**Review by uthmanmoh — APPROVED** [review]

**Pull request merged**